[dev-docs] Add prebuild to check for .env vars

hariria opened the pull request:
### Description
Same as title

### Checklist

- Do all Lints pass?
  - [x] Have you ran `pnpm spellcheck`?
  - [x] Have you ran `pnpm fmt`?
  - [x] Have you ran `pnpm lint`?

gregnazario reviewed (commented):
Can we set the variable with a default, so users don't have to run `prebuild`?

Either that or have one command that will set up everything regardless of your platform

hariria commented:
Spent a bunch of time looking at alternatives. There seems to be no way to set a default `.env` without committing it and thus not ignoring it. I have added a `.env.example` though which should hopefully help. 

Also, to clarify the problem, I think the error is actually quite good, it's just kind of hidden when you run build from the root. 

```text
Error: Origin is not defined in docs.config.js. Ensure it is part of your .env file for your project by adding something like `NEXT_PUBLIC_ORIGIN="http://localhost:3030"`
```

Anyways, I rewrote the README.md to try and make it super clear and hopefully now that the `.env.example` is there, it shouldn't be a huge issue anymore.
